composeApp/src/commonMain/kotlin/com/swmansion/routinetracker/screen/CreateTaskScreen.kt

```diff
@@ -0,0 +1,212 @@
+package com.swmansion.routinetracker.screen
+
+import androidx.compose.foundation.layout.*
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
+import androidx.compose.runtime.collectAsState
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.MutableCreationExtras
+import androidx.lifecycle.viewmodel.compose.viewModel
+import androidx.navigation.NavController
+import com.mohamedrejeb.calf.ui.timepicker.AdaptiveTimePicker
+import com.mohamedrejeb.calf.ui.timepicker.AdaptiveTimePickerState
+import com.mohamedrejeb.calf.ui.timepicker.rememberAdaptiveTimePickerState
+import com.swmansion.routinetracker.di.LocalAppContainer
+import com.swmansion.routinetracker.navigation.CreateRoutine
+import com.swmansion.routinetracker.viewmodel.CreateRoutineViewModel
+import org.jetbrains.compose.resources.painterResource
+import routinetracker.composeapp.generated.resources.Res
+import routinetracker.composeapp.generated.resources.ic_back
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun CreateTaskScreen(navController: NavController) {
+    val routineBackStackEntry =
+        remember(navController) { navController.getBackStackEntry(CreateRoutine) }
+    val viewModel: CreateRoutineViewModel =
+        viewModel(
+            routineBackStackEntry,
+            factory = CreateRoutineViewModel.Factory,
+            extras =
+                MutableCreationExtras().apply {
+                    set(
+                        CreateRoutineViewModel.DATA_REPOSITORY_KEY,
+                        LocalAppContainer.current.repository,
+                    )
+                },
+        )
```

Comment:
Elsewhere, instead of such a property, we use a parameter with a default value. This is helpful for testing, enabling you to pass in a mock viewmodel. Let’s do the same here for consistency.